tests/resources/application.yml

```diff
@@ -160,6 +159,11 @@ test:
 
 cis_mode: false
 
+version:
+  error: "uninitialized"
+  default: "0xF"  # this tests alphanumeric defaults
```

Comment:
It's something Malcolm had hinted at, so I just did it now, rather than re-implementing everything later.  As implemented now, it does not impact internal management of versions, where the default is the integer `-1`.  We don't have to tell anyone this is supported, but it is.